spec/models/purchase_spec.rb

```diff
@@ -0,0 +1,41 @@
+require 'rails_helper'
+
+describe Purchase do
+
+  before(:each) do
+    Purchase.create(email: "me@example.com", item_id: 2)
+    @purchase = Purchase.first
+  end
+
+  context "purchase is valid" do
+    it "has an email" do
+
+      expect(@purchase.email).to eq("me@example.com")
+    end
+
+    it "has an item_id" do
+
+      expect(@purchase.item_id).to eq(2)
+    end
+  end
+
+  context "item is invalid" do
+    it "doesn't save without an email" do
+      purchase = create_invalid_purchase(item_id: 1)
+
+      expect(purchase.errors).to include(:email)
+    end
+
+    it "doesn't save without an item id" do
+      purchase = create_invalid_purchase(email: "us@example.com")
+
+      expect(purchase.errors).to include(:item_id)
+    end
+  end
+
+  private
+
+  def create_invalid_purchase(purchase_details)
```

Comment:
This method is not title appropriately, there is nothing about it that creates an invalid purchase. The validity of a purchase is defined by the attributes it's created with. I'd prefer to just see `Purchase.create(your_purchase_data)` in each spec. There is nothing about pulling that into a method that increases readability.